### ISSUE TITLE
Custom slugs for study detail URLs instead of UUIDs

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#624]
# TL;DR
Allow accessing the participant-facing study detail page using a study slug rather than the UUID.

# Narrative
As a researcher I'd like my study detail page to be prettier for parent-facing materials and more conducive to SEO. 

# Acceptance Criteria

- [ ] Slug field added to study model
- [ ] Slug is editable on study edit form with uniqueness enforced and a clear message if there's a collision
- [ ] Button to create slug from study name on study edit form
- [ ] Slug-based URL is displayed on the researcher-facing study detail page as the link
- [ ] Slug-based URL is used everywhere we link to the study detail page (e.g., study list, announcement emails)
- [ ] Tests are adjusted to use this URL and pass

# Implementation Notes
We will need to continue to support the UUID-based URLs at least temporarily since links will have been sent out to families. This could be done via a migration that initially sets the study slug to a string representation of the UUID, or by having both URL paths for now. 

Note that the actual participation/preview views will still need to use the UUID for compatibility with the experiment runner.
